ui/src/lib/yjs/useWorkflowTabs.ts

```diff
@@ -18,17 +18,24 @@ export default ({
     [currentWorkflowId],
   );
 
+  const [workflowNames, setWorkflowsNames] = useState(
+    rawWorkflows.map((w) => ({ id: w.id, name: w.name })),
+  );
   // This works as a semi-static base for the rest of the state in this hook.
   // Without this state (aka using rawWorkflows directly), performance drops
   // due to the state updating on every change to a node (which is a lot)
-  const workflows = useMemo(
-    () =>
-      rawWorkflows.filter(isDefined).map((w2) => ({
-        id: w2.id as string,
-        name: w2.name as string,
-      })),
-    [rawWorkflows.length], // eslint-disable-line react-hooks/exhaustive-deps
-  );
+  useEffect(() => {
+    if (rawWorkflows.length !== workflowNames.length) {
+      setWorkflowsNames(rawWorkflows.map((w) => ({ id: w.id, name: w.name })));
+    }
+  }, [rawWorkflows.length, workflowNames.length]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
I feel like its better to keep updating consistent, which means handling workflow add/removal should happen in the same way as the rename in the Editor hooks.